httomo_backends/methods_database/packages/backends/httomolibgpu/supporting_funcs/recon/algorithm.py

```diff
@@ -148,6 +153,94 @@ def _calc_memory_bytes_FBP(
     return (tot_memory_bytes, fixed_amount)
 
 
+
+def _calc_memory_bytes_LPRec(
+    non_slice_dims_shape: Tuple[int, int],
+    dtype: np.dtype,
+    **kwargs,
+) -> Tuple[int, int]:
+    angles_tot = non_slice_dims_shape[0]
+    DetectorsLengthH = non_slice_dims_shape[1]
+
+    # calculate the output shape
+    output_dims = __calc_output_dim_recon(non_slice_dims_shape, **kwargs)
+
+    #input and and output slices
+    in_slice_size = np.prod(non_slice_dims_shape) * dtype.itemsize
+    out_slice_size = np.prod(output_dims) * dtype.itemsize
+
+    # interpolation kernels
+    # grid_size = np.prod(DetectorsLengthH * DetectorsLengthH) * np.float32().itemsize
+    # phi = grid_size
+
+    n = DetectorsLengthH
+
+    odd_horiz = False
+    if (n % 2) != 0:
+        n = n - 1  # dealing with the odd horizontal detector size
+        odd_horiz = True
+
+    eps = 1e-4  # accuracy of usfft
+    mu = -np.log(eps) / (2 * n * n)
+    m = int(
+        np.ceil(
+            2
+            * n
+            * 1
+            / np.pi
+            * np.sqrt(
+                -mu * np.log(eps)
+                + (mu * n) * (mu * n) / 4
+            )
+        )
+    )
+
+    data_c_size = np.prod(0.5 * angles_tot * n) * np.complex64().itemsize
+
+    fde_size = (
+        0.5 * (2 * m + 2 * n) * (2 * m + 2 * n)
+    ) * np.complex64().itemsize
+
+    c1dfftshift_size = (
+        n * np.int8().itemsize
+    )
+
+    c2dfftshift_slice_size = (
+        np.prod(4 * n * n) * np.int8().itemsize
+    )
+
+    theta_size = angles_tot * np.float32().itemsize
+    filter_size = (n // 2 + 1) * np.float32().itemsize
+    freq_slice = angles_tot * (n + 1) * np.complex64().itemsize
+    fftplan_size = freq_slice * 2
+
+    phi_size = n * n * np.float32().itemsize
+
+    # We have two fde arrays and sometimes need double fde.
+    max_memory_per_slice = max(data_c_size + 2 * fde_size, 3 * fde_size)
+
+    tot_memory_bytes = int(
+        in_slice_size
+        + out_slice_size
+        + max_memory_per_slice
+    )
+
+    fixed_amount = int(
+        fde_size
+        + data_c_size
+        + theta_size
+        + fftplan_size
```

Comment:
Should we account for the plan within `tot_memory_bytes` ?  Although it is reused in a loop, I reckon it will peak on the first slice iteration?